terratorch/tasks/segmentation_tasks.py

```
@@ -106,8 +108,21 @@ def __init__(
         self.tiled_inference_parameters = tiled_inference_parameters
         self.aux_loss = aux_loss
         self.aux_heads = aux_heads
-        self.model_factory = MODEL_FACTORY_REGISTRY.build(model_factory)
+
+        if model is not None and model_factory is not None:
+            logger.warning("A model_factory and a model was provided. The model_factory is ignored.")
+        assert model is not None or model_factory is not None, \
```

can we raise an error instead? assertions can be switched off